File: nats/src/test/groovy/io/micronaut/nats/docs/quickstart/ProductClient.java
package io.micronaut.nats.docs.quickstart;

// tag::imports[]
import io.micronaut.nats.annotation.NatsClient;
import io.micronaut.nats.annotation.Subject;
import io.micronaut.messaging.annotation.Body;
// end::imports[]

// tag::clazz[]
@NatsClient // <1>
public interface ProductClient {

    @Subject("abc") // <2>
    void send(@Body byte[] data); // <3>

}
// end::clazz[]

File: docs-examples/example-python/src/test/python/micronaut/nats/docs/consumer/custom/type/ProductClient.py
from abc import ABC, abstractmethod
from typing import Annotated

from java.lang import Long
from micronaut.context.annotation import Requires
# tag::imports[]
from micronaut.messaging.annotation import MessageHeader
from micronaut.nats.annotation import NatsClient, Subject
# end::imports[]


@Requires(property="spec.name", value="ProductInfoSpec")
# tag::clazz[]
@NatsClient
@MessageHeader(name="x-product-sealed", value="true")  # <1>
@MessageHeader(name="productSize", value="large")
class ProductClient(ABC):

    @Subject("product")
    @MessageHeader(name="x-product-count", value="10")  # <2>
    @MessageHeader(name="productSize", value="small")
    @abstractmethod
    def send(self, data: bytes) -> None:
        ...

    @Subject("product")
    @abstractmethod
    def send_with_headers(self, product_size: Annotated[str | None, MessageHeader("productSize")],  # <3>
                          count: Annotated[Long, MessageHeader("x-product-count")],
                          data: bytes) -> None:
        ...
# end::clazz[]
